src/main/java/com/datastax/loader/CqlDelimLoad.java

```diff
@@ -490,27 +439,39 @@ private boolean setup()
         throws IOException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException,
                CertificateException, UnrecoverableKeyException {
         // Connect to Cassandra
-        PoolingOptions pOpts = new PoolingOptions();
-        pOpts.setMaxConnectionsPerHost(HostDistance.LOCAL, 8);
-        pOpts.setCoreConnectionsPerHost(HostDistance.LOCAL, 8);
-        Cluster.Builder clusterBuilder = Cluster.builder()
-            .addContactPoint(host)
-            .withPort(port)
-            //.withCompression(ProtocolOptions.Compression.LZ4)
-            .withPoolingOptions(pOpts)
-            .withLoadBalancingPolicy(new TokenAwarePolicy( DCAwareRoundRobinPolicy.builder().build()))
-            ;
-
-        if (null != username)
-            clusterBuilder = clusterBuilder.withCredentials(username, password);
-        if (null != truststorePath)
-            clusterBuilder = clusterBuilder.withSSL(createSSLOptions());
-
-        cluster = clusterBuilder.build();
-        if (null == cluster) {
-            throw new IOException("Could not create cluster");
-        }
-        Session tsession = cluster.connect();
+        Session tsession = null;
+        try {
+            PoolingOptions pOpts = new PoolingOptions();
+            pOpts.setMaxConnectionsPerHost(HostDistance.LOCAL, 8);
+            pOpts.setCoreConnectionsPerHost(HostDistance.LOCAL, 8);
+            Cluster.Builder clusterBuilder = Cluster.builder()
+                    .addContactPoint(host)
+                    .withPort(port)
+                    //.withCompression(ProtocolOptions.Compression.LZ4)
+                    .withPoolingOptions(pOpts)
+                    .withLoadBalancingPolicy(new TokenAwarePolicy(DCAwareRoundRobinPolicy.builder().build()));
+
+            if (null != username)
+                clusterBuilder = clusterBuilder.withCredentials(username, password);
+            if (null != truststorePath)
+                clusterBuilder = clusterBuilder.withSSL(createSSLOptions());
+
+            cluster = clusterBuilder.build();
+            if (null == cluster) {
+                throw new IOException("Could not create cluster");
+            }
+            tsession = cluster.connect();
+        }
+        catch (IllegalArgumentException e){
+            System.err.println("Could not connect to the cluster, check your hosts");
+            //e.printStackTrace();
+            System.exit(0);
```

Comment:
System.exit should really be avoided, why not throw an IOException like on line 461?

Same thing below